cni/pkg/nodeagent/server.go

```diff
@@ -66,6 +71,36 @@ type Server struct {
 	cniServerStopFunc func()
 }
 
+// runAsNs executes the function in the given network namespace or the current one if netNs is nil.
+func runAsNs(netNs netns.NetNS, f func() error) error {
```

Comment:
really, all we need is a closure that gets the FD from the known `/proc` path and does `netns.Do`, and we just call that closure from inside `hostXXX` funcs.

Reply:
This is how it worked in the PR before I started iterating on it.
I will restore the logic with netns.WithNetNSPath(path, f) , it will take care of lifecycle of the fd